operator/v1/types.go

```diff
@@ -52,6 +52,43 @@ type ClusterManagerSpec struct {
 	// +optional
 	// +kubebuilder:default={mode: Default}
 	DeployOption DeployOption `json:"deployOption,omitempty"`
+
+	// Detached includes configurations we needs for clustermanager in the detached mode.
+	// +optional
+	Detached DetachedClusterManagerConfiguration `json:"detached,omitempty"`
+}
+
+// DetachedClusterManagerConfiguration represents customized configurations we need to set for clustermanager in the detached mode.
+// After a detached mode deployment, the webhook-servers will not be feasible until the user exposes the webhook-servers in the management cluster manually. And the way to do that is depending on the platform and user's preference.
+// For example, if we are using a OCP platform, we could use Route to expose the webhook-servers.
+type DetachedClusterManagerConfiguration struct {
+	// RegistrationWebhookConfiguration represents the customized webhook-server configuration of registration.
+	// +optional
+	RegistrationWebhookConfiguration WebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
+
+	// WorkWebhookConfiguration represents the customized webhook-server configuration of work.
+	// +optional
+	WorkWebhookConfiguration WebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
+}
+
+// WebhookConfiguration has two properties: Address and Port.
+// The registration and work provide webhook-server by using [generic-admission-server](https://github.com/openshift/generic-admission-server).
+// And this lib has simplified the authentication part of a webhook-server, and the only thing we need to do is set up a Service for the respective APIService automatically.
+type WebhookConfiguration struct {
+	// Address represents the address of a webhook-server. It could be in IP format or fqdn format.
+	// If Address is in IP format, operator will generate a headless Service pointing to the IP.
```

Comment:
It does not have to be doced whether headless or externalName is used here.

Reply:
Fixed.